syntax = "proto3";

service ShowMessage {
    rpc Show(InputMessage) returns (OutputMessage) {}
}

message InputMessage {
    string value = 1;
}

message OutputMessage {
    string value = 1;
}

message TwitterCredentials {
    string consumer_key = 1;
    string consumer_secret = 2;
    string access_token = 3;
    string token_secret = 4;
}

message TwitterInputMessage {
    TwitterCredentials credentials = 1;
    string languages = 2;
    string query = 3;
    int64 time_limit = 4;
    int64 msg_limit = 5;
    string product = 6;
    string environment = 7;
    string from_date = 8;
    string to_date = 9;
    int64 messages_per_request = 10;
    int64 max_requests_limit = 11;
    string db_name = 12;
}

message TwitterOutputMessage {
    string value = 1;
}

service SentimentConsensusAnalysis {
    rpc ConsensusAnalysis(InputMessage) returns (OutputMessage) {}
}

service TwitterHistoricalAnalysis {
    rpc HistoricalAnalysis(TwitterInputMessage) returns (OutputMessage) {}
    rpc HistoricalAnalysisToDatabase(TwitterInputMessage) returns (OutputMessage) {}
}

service TwitterStreamAnalysis {
    rpc StreamAnalysis(TwitterInputMessage) returns (OutputMessage) {}
}